addons/website_sale_digital/controllers/main.py

```diff
@@ -57,11 +57,7 @@ def portal_order_page(self, order_id=None, **post):
     ], type='http', auth='public')
     def download_attachment(self, attachment_id):
         # Check if this is a valid attachment id
-        attachment = request.env['ir.attachment'].sudo().search_read(
-            [('id', '=', int(attachment_id))],
-            ["name", "datas", "mimetype", "res_model", "res_id", "type", "url"]
-        )
-
```

Comment:
The `_get_stream_from` method expect `attachement` to be a record and `search_read` always returns a dictionnary.

Reply:
This kind of explanation should be in your commit message and not on the PR ;).

Reply:
thank you :)